src/server/scripts/Kalimdor/HallsOfOrigination/boss_isiset.cpp

```diff
@@ -0,0 +1,1071 @@
+                return;
+
+            mirrorImageDied = true;
+            
+            // Handle despawning
+            DummyEntryCheckPredicate pred;
+            summons.DoAction(ACTION_IMAGES_SET_PASSIVE, pred);
+            events.ScheduleEvent(EVENT_DESPAWN_IMAGES, 2400);
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            //instance->SendEncounterUnit(ENCOUNTER_FRAME_SET_COMBAT_RES_LIMIT, ?me?, 1);
+            instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me, 0);
+            _JustDied();
+            Talk(SAY_DEATH);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            events.Update(diff);
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_START_ATTACK:
+                    instance->SendEncounterUnit(ENCOUNTER_FRAME_UPDATE_PRIORITY, me, 1);
+                    me->SetReactState(REACT_AGGRESSIVE);
+                    DoStartMovement(me->GetVictim());
+                    break;
+                case EVENT_ASTRAL_RAIN:
+                    DoCast(me, SPELL_ASTRAL_RAIN_CONTROLLER);
+                    events.ScheduleEvent(EVENT_ASTRAL_RAIN, urand(15,19)*IN_MILLISECONDS);
+                    break;
+                case EVENT_ASTRAL_FAMILIAR:
+                    DoCast(me, SPELL_ASTRAL_FAMILIAR_CONTROLLER);
+                    events.ScheduleEvent(EVENT_ASTRAL_FAMILIAR, urand(26,30)*IN_MILLISECONDS);
+                    break;
+                case EVENT_VEIL_OF_SKY:
+                    DoCast(me, SPELL_MANA_SHIELD_CONTROLLER);
+                    events.ScheduleEvent(EVENT_VEIL_OF_SKY, urand(20,25)*IN_MILLISECONDS);
+                    break;
+                case EVENT_SUPERNOVA:
+                    DoCast(me, SPELL_SUPERNOVA);
+                    Talk(SAY_SUPERNOVA);
+                    Talk(SAY_SUPERNOVA_WARNING);
+                    events.ScheduleEvent(EVENT_SUPERNOVA, urand(45,50)*IN_MILLISECONDS);
+                    break;
+                case EVENT_IMAGES_ATTACK:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_SET_AGGRESSIVE, pred);
+                    break;
+                }
+                case EVENT_DESPAWN_IMAGES:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_DESPAWN, pred);
+                    instance->SetData(DATA_ISISET_PHASE, instance->GetData(DATA_ISISET_PHASE) + 1);
+                    me->RemoveAurasDueToSpell(SPELL_ASTRAL_SHIFT);
+                    RescheduleEvents();
+                    break;
+                }
+                default:
+                    break;
+                }
```

Comment:
Missing a if (me->HasUnitState(UNIT_STATE_CASTING)) return; here

Reply:
There is one before event execute switch